proposals/profiles/Overview.md

```diff
@@ -51,6 +51,33 @@ For example, with respect to current Wasm as well as ongoing proposal work, the
 In each case, a related profile is expected to affect all instructions and constructs associated with a respective feature.
 
 
+#### Goals
+
+1. *Well-specified subsets.* Provide new Wasm ecosystems that cannot support the full language with suitable subsets, which would otherwise be forced to invent them on their own.
+
+2. *Minimised fragmentation.* Maximise compatibility between across diverse ecosystems by aligning them and keeping the number of profiles to a minimum. (Also, minimise complexity and test matrix for engines or tools that want to be configurable for multiple profiles.)
+
+3. *Durability.* The set of available profiles should remain fairly stable over time, such that ecosystems rarely if ever need to reconsider their choice.
+
+
+#### Non-Goals
+
+1. *Versioning.* Profiles are _not_ intended to distinguish new vs old versions of the language – that is an independent dimension for which we already have have version numbers. Rather, we fully expect that new versions of the language will extend existing profiles. Profiles are meant to codify _permanent_ use case distinctions, not temporary implementation deviations.
+
+2. *Feature detection.* Profiles are directed at ecosystem creators, not users of the language. While it may make sense to provide programmatic means for certain applications to become portable even across incompatible ecosystems, such means are outside the scope of this proposal.
+
+3. *Alternate semantics.* Profiles can only _restrict_ the language, they are never intended to _change_ it, e.g., by introducing alternative features or alternative behaviours. Language changes would introduce significantly larger complications for ecosystems, implementations, and users. In particular, they would preclude the existence of a "full profile" that includes everything, and targetting which allows implementations to remain profile-agnostic.
```

Comment:
> I wonder if it makes sense to have the "full profile" define the relaxed-simd opcodes with set nondeterminism, and then have relaxed-math mode be a subset which introduces list nondeterminism?

Yes, semantically that would work. Editorially, the list semantics would still have to be part of the main spec, but practically overridden by set determinism, unless that is disabled by a profile.

In general, the main body of the spec has to specify the union of all possible behaviours. So unfortunately, we cannot technically shove out non-determinism there.

Reply:
> A subtlety here is that relaxed-math mode is proposed to be framed such that deterministic mode is the prominent mode in the spec, and relaxed mode is an alternate superset of it.

Personally, I have a somewhat strong opinion that we should frame things in terms of a "full/default" profile and a (clearly marked alternative) "deterministic/strict" profile/mode. The safe thing for users to do is to write/compile/produce their program anticipating the non-deterministic semantics, because then moving to a "deterministic/strict" implementation is safe. If we message that the deterministic semantics is the "normal" one, and this bleeds into documentation or tooling, I worry that we'd be creating a hazard for users.

> I wonder if it makes sense to have the "full profile" define the relaxed-simd opcodes with set nondeterminism, and then have relaxed-math mode be a subset which introduces list nondeterminism?

Given the intended use-cases of `qfma` I think even the full profile is going to need list non-determinism.